cmd/kubeadm/app/cmd/join.go

```diff
@@ -342,6 +342,9 @@ func newJoinData(cmd *cobra.Command, args []string, opt *joinOptions, out io.Wri
 
 	// if not joining a control plane, unset the ControlPlane object
 	if !opt.controlPlane {
+		if len(opt.externalcfg.ControlPlane.CertificateKey) > 0 {
```

Comment:
i think `ControlPlane` can be `nil` here?

Reply:
If `--control-plane` is false, we are ignoring all the configurations under `opt.externalcfg.ControlPlane` (bot `certificateKey` and `localAPIEndpoint`)

So IMO the check and the message should be made more generic e.g
```
if !opt.controlPlane {
  if opt.externalcfg.ControlPlane!=nil {
     klog.Warningf("[preflight] WARNING: JoinControlPane.controlPlane settings will be ignored when %s flag is not set.", options.ControlPlane)
  }
}
```